File: Web2-Microservices/RACSMicroservice/Models/RentACarService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RACSMicroservice.Models
{
    public class RentACarService
    {
        public int RentACarServiceId { get; set; }
        public string Name { get; set; }
        public string AdminId { get; set; }
        public Address Address { get; set; }
        public string About { get; set; }
        public ICollection<Branch> Branches { get; set; }
        public ICollection<Car> Cars { get; set; }
        public byte[] LogoUrl { get; set; }
        public ICollection<RentCarServiceRates> Rates { get; set; }

        public RentACarService()
        {
            Branches = new HashSet<Branch>();
            Cars = new HashSet<Car>();
            Rates = new HashSet<RentCarServiceRates>();
        }
    }
}

File: Web-SPA/src/app/entities/rent-a-car-service.ts
import { Destination } from './destination';
import { Address } from './address';
import { Car } from './car';

export class RentACarService {

    id: number;
    adminId: number;
    name: string;
    address: Address;
    about: string;
    promoDescription: Array<string>;
    branches: Array<RentACarService>;
    averageRating: number;
    rates: Array<number>;
    cars: Array<Car>;
    logoUrl: string;

    constructor(name: string, address: Address) {
        this.name = name;
        this.address = address;
        this.about = '';
        this.promoDescription = new Array<string>();
        this.branches = new Array<RentACarService>();
        this.averageRating = 0.0;
        this.rates = new Array<number>();
        this.cars = new Array<Car>();
        this.logoUrl = '';
    }
}
